src/core/attribution/markovChain.test.js

```diff
@@ -184,49 +244,373 @@ describe("core/attribution/markovChain", () => {
         [0.5, 0, 0.25, 0, 0.25],
         [0.5, 0.25, 0, 0.25, 0],
       ]);
-      const result = await findStationaryDistribution(chain, {
-        maxIterations: 255,
-        convergenceThreshold: 1e-7,
-        verbose: false,
-        yieldAfterMs: 1,
-      });
+      const alpha = 0;
+      const seed = uniformDistribution(chain.length);
+      const initialDistribution = uniformDistribution(chain.length);
+      const result = await findStationaryDistribution(
+        chain,
+        seed,
+        alpha,
+        initialDistribution,
+        {
+          maxIterations: 255,
+          convergenceThreshold: 1e-7,
+          verbose: false,
+          yieldAfterMs: 1,
+        }
+      );
 
       expect(result.convergenceDelta).toBeLessThanOrEqual(1e-7);
-      validateConvegenceDelta(chain, result);
+      validateConvegenceDelta(chain, seed, alpha, result);
 
-      expectStationary(chain, result.pi);
+      expectStationary(chain, seed, alpha, result.pi);
       const expected = new Float64Array([1 / 3, 1 / 6, 1 / 6, 1 / 6, 1 / 6]);
       expectAllClose(result.pi, expected);
     });
 
+    it("finds a the same stationary distribution regardless of initialDistribution", async () => {
+      // Node 0 is the "center"; nodes 1 through 4 are "satellites". A
+      // satellite transitions to the center with probability 0.5, or to a
+      // cyclically adjacent satellite with probability 0.25 each. The
+      // center transitions to a uniformly random satellite.
+      const chain = sparseMarkovChainFromTransitionMatrix([
+        [0, 0.25, 0.25, 0.25, 0.25],
+        [0.5, 0, 0.25, 0, 0.25],
+        [0.5, 0.25, 0, 0.25, 0],
+        [0.5, 0, 0.25, 0, 0.25],
+        [0.5, 0.25, 0, 0.25, 0],
+      ]);
+      const alpha = 0.1;
+      const seed = uniformDistribution(chain.length);
+      const initialDistribution1 = indicatorDistribution(chain.length, 0);
+      const initialDistribution2 = indicatorDistribution(chain.length, 1);
+
+      const result1 = await findStationaryDistribution(
+        chain,
+        seed,
+        alpha,
+        initialDistribution1,
+        {
+          maxIterations: 255,
+          convergenceThreshold: 1e-7,
+          verbose: false,
+          yieldAfterMs: 1,
+        }
+      );
+
+      const result2 = await findStationaryDistribution(
+        chain,
+        seed,
+        alpha,
+        initialDistribution2,
+        {
+          maxIterations: 255,
+          convergenceThreshold: 1e-7,
+          verbose: false,
+          yieldAfterMs: 1,
+        }
+      );
+
+      expectAllClose(result1.pi, result2.pi);
+    });
+
+    it("finds a non-degenerate stationary distribution with seed and non-zero alpha", async () => {
```

Comment:
Not sure. @mzargham can answer maybe.

Reply:
a probability distribution is degenerate if its support is lower dimensional. In this case that would mean not every node gets a score. 
https://en.wikipedia.org/wiki/Degenerate_distribution
I don't think this test checks that though, it  looks like it just checks that a stationary distribution was found. 

Some nodes might get arbitrarily small amounts of cred but technically no node in a connected graph, even with personalized page rank should get strictly zero cred. I don't think we need to check for non-degeneracy since numberically we'll get 'might as well be 0' values. My suggestion is change the wording on the test.

Reply:
Ok, I removed "non-degenerate" from the test description.

Reply:
Connected Graph is a property of the graph whereas irreducible chain is a property of the Markov Process. Graph properties are more general as they do not depend on the weights or on any particular graph process for that matter. Markov chains are mixing processes defined over graphs; a Markov process defined over a digraph which is not strongly connected will be reducible to another Markov process which is defined over a subgraph which is strongly connected. None of this is particularly relevant to the implementation, but I am inclined to nit pick in the graph theory and dynamical systems domain.

Reply:
Agreed on just changing the name of the test.

Reply:
Sure, but a graph on two nodes with a single edge is always connected,
while the chain is irreducible iff the edge is of nonzero weight.

Reply:
Now we're just getting into semantics, because a graph with an edge weight which is zero breaks the graph abstraction; in most conventions that edge is now not considered present, and the graph is disconnected. One of the reasons I love math is that we wouldn't need to have this discussion if we were working with formalisms instead of natural language.

Reply:
Also, reference case for why comment chains shouldn't absorb too much cred. As much as I enjoy math banter, especially when graphs are involved, I am quite sure we are not meaningfully contributing to the project.

Reply:
But if “[g]raph properties are more general as they do not depend on the
weights”, then surely connectedness of the digraph cannot depend on
whether the weight is zero or not?

> we wouldn't need to have this discussion if we were working with
> formalisms instead of natural language

Indeed, and yet I think that we’re on the same page.

Reply:
I got sucked in and made a little bit of LaTeX with my preferred formalism. This further extends to defining mixing processes on the graph which are actually dynamical systems. We can normalize, add self loops, at driving functions etc and before long arrive at many other concepts: Laplacians, Markov Chains and PageRank, etc.
![image](https://user-images.githubusercontent.com/10465438/56476935-7b4a2d00-6454-11e9-8da5-40368b184a49.png)

These formalism are generally introduced with the phrase "without loss of generality" which in the academic literature basically means: "i know you can make an example that violates this but then I can map that example onto another case that respects this...". A common example being assumptions that graphs are connected because the disconnected graphs can then be analyzed as multiple separate connected graphs.

I was embroiled in Network Science academic space for more years than I care to count. I have scars.